skops/card/_model_card.py

```diff
@@ -1315,3 +1315,36 @@ def render(self) -> str:
             sections inserted.
         """
         return "\n".join(self._generate_card())
+
+    def _iterate_key_section_content(
+        self,
+        data: dict[str, Section],
+        level: int = 0,
+    ):
+        for key, val in data.items():
+            if not getattr(val, "visible", True):
+                continue
+
+            title = val.title
+            yield title, level
+
+            if val.subsections:
+                yield from self._iterate_key_section_content(
+                    val.subsections,
+                    level=level + 1,
+                )
+
+    def create_toc(self) -> str:
+        """Create a table of contents for the model card.
```

Comment:
I think we need some clarification on this method. As a user, I might think that calling this method will "create a toc" in the sense that it will be automatically added to the model card when it is saved or rendered. The text should therefore clarify that this method only returns the toc for convenience but doesn't modify the model card.

Reply:
Maybe instead of this design, we should have something line

```
	def include_toc(visible/expanded=True):
```

which adds the ToC during render time, and also allows it to be hidden/collapsed.

Reply:
The primary use case I had in mind when thinking about this is if I'm interactively building up my model card. I would like to get a quick overview of what is and isn't there. Adding a toc to the final model card would not help, as it would require me to render and read the md, which is cumbersome.

Now it could be interesting to _also_ have a method to add the toc to the card. But as mentioned elsewhere, at least on GH and HF Hub, the toc is created automatically, so it's not necessary there. We could still add it, but I wouldn't use a method, but an init argument (since the toc should be lazily created when the card is rendered).

Reply:
> We could still add it, but I wouldn't use a method, but an init argument (since the toc should be lazily created when the card is rendered).

I agree, if we're going to have it on the model card we should just add an argument.

Reply:
in that case, why not a read-only property `toc` which would return the current `toc`?

Reply:
What would be the advantage of having a property compared to a method? I'd slightly prefer a method here, e.g. in case we want to later add arguments to it.

Reply:
I'm not sure what arguments you'd pass here, but if you want a method, then it should be `get_toc`

Reply:
Not sure either, but at least that leaves the door open. I'm fine with `get_toc`.

Reply:
@lazarust could you please rename the method to `get_toc`?